homeassistant/components/spc/__init__.py

```diff
@@ -62,15 +77,31 @@ async def async_upate_callback(spc_object):
         _LOGGER.error("Failed to load area/zone information from SPC")
         return False
 
+    device_registry = dr.async_get(hass)
+    device_registry.async_get_or_create(
+        config_entry_id=entry.entry_id,
+        identifiers={(DOMAIN, spc.info["sn"])},
+        manufacturer="Vanderbilt",
+        name=spc.info["sn"],
+        model=spc.info["type"],
+        sw_version=spc.info["version"],
+    )
+
     # add sensor devices for each zone (typically motion/fire/door sensors)
     hass.async_create_task(
-        discovery.async_load_platform(hass, Platform.BINARY_SENSOR, DOMAIN, {}, config)
+        discovery.async_load_platform(
```

Comment:
You should use `hass.config_entries.async_forward_entry_setups` instead to load platforms.

Reply:
Thank you for the pointers!